src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -140,7 +139,8 @@ function SpectralTransform(
     end
     
     # SCRATCH MEMORY FOR FOURIER NOT YET LEGENDRE TRANSFORMED AND VICE VERSA
-    scratch_memory_north = zeros(Complex{NF}, nfreq_max, nlayers, nlat_half)
+    # north is converted to array type now for help with making the FFT plans
+    scratch_memory_north = ArrayType(zeros(Complex{NF}, nfreq_max, nlayers, nlat_half))
     scratch_memory_south = zeros(Complex{NF}, nfreq_max, nlayers, nlat_half)
```

Comment:
This one also needs to be `ArrayType` , same for the `scratch_memory_column_north` below

Reply:
Everything gets converted to CuArrays at the end of the constructor, only the north scratch memory is needed for the fft planning so I didn't explicitly convert anything else but I can do happily.

Reply:
Ah right, it gets automatically converted by the constructor, right? I missed that.